packages/payment-processor/src/payment/erc777-utils.ts

```diff
@@ -0,0 +1,226 @@
+import { ContractTransaction, Signer, providers, BigNumberish, BigNumber } from 'ethers';
+
+import { ClientTypes, ExtensionTypes, PaymentTypes } from '@requestnetwork/types';
+import { getPaymentNetworkExtension } from '@requestnetwork/payment-detection';
+
+import { getNetworkProvider, getProvider, validateRequest, MAX_ALLOWANCE } from './utils';
+import Token from '@superfluid-finance/sdk-core/dist/module/Token';
+import { getSuperFluidFramework } from './erc777-stream';
+import Operation from '@superfluid-finance/sdk-core/dist/module/Operation';
+
+/**
+ * Gets the underlying token address of an ERC777 currency based request
+ * @param request the request that contains currency information
+ * @param provider the web3 provider. Defaults to Etherscan
+ */
+export async function getRequestUnderlyingToken(
```

Comment:
If this is specific to SF, the return type should be a `SuperToken`, but can't we use a generic version of this method? If it works with a `Token` return type, it may mean you don't need the sf framework.

Reply:
@yomarion by the way, using the SF functions could enable us to batch them in SF batch call if it makes sense at some point